### ISSUE TITLE
x11docker --install fails on macOS 10.15.4

### ISSUE THREAD

**Issue by BackOrder** [mviereck/x11docker#239]
```zsh
⇒  sudo ./x11docker --install
Password:
./x11docker: line 8334: exec: {FDstderr}: not found
```

```zsh
⇒  curl -fsSL https://raw.githubusercontent.com/mviereck/x11docker/master/x11docker | sudo bash -s -- --update
bash: line 8334: exec: {FDstderr}: not found
```

**Comment by mviereck** [comment]
x11docker does not run on macOS. From README.md:
> x11docker runs on Linux and (with some setup and limitations) on MS Windows. x11docker is not adapted to run on macOS except in a Linux VM.

**Comment by BackOrder** [comment]
Ah, how about a label **wontfixitself** ? :D 

What would it take to make it work on macOS? I can successfully run x11 dockers on macOS. Perhaps I could check a bit with you how to make some magic happen.

**Comment by eine** [comment]
@BackOrder, see #15 and #94.

**Comment by mviereck** [comment]
> What would it take to make it work on macOS?

I fear the effort. x11docker is supposed to run on almost every Linux system (including Cygwin on MS Windows). It was a pita to make all commands compatible across a lot of systems and checking it out in several VMs. 

macOS is quite different in command details, and Apple does not provide a free developer VM. And, last but not least, it is closed source.
So I once decided not to support macOS.

> Perhaps I could check a bit with you how to make some magic happen.

If you decide to fork x11docker and to maintain your own macOS version, I could provide some help. Note that I would not maintain it myself and would not merge it in x11docker.

Basically you would have to run it over and over again and fix every single failing command. Always run with option `-D, --debug` to cause a stop on each failing command.

The code path would be similar to MS Windows setups because Docker on macOS is similar to the MobyVM solution on MS Windows.

**Comment by BackOrder** [comment]
Thank you, both @mviereck and @eine !

**Comment by karawitan** [comment]
Hi !

Hopefully x11docker might work on OSX (Apple Silicon) soon:

https://github.com/karawitan/x11docker/blob/osx-monterey/README-OSX.txt

**Comment by mviereck** [comment]
Interesting!
If you have questions on the intention of failing commands, feel free to ask.
I've set the repository on my watchlist, so I'll see your changes.
